test/suites/smoke/test-polkadot-decoding.ts

```diff
@@ -107,19 +131,25 @@ describeSuite({
                 // 2. Extract the module, fn and params keys
                 const { moduleKey, fnKey, paramsKey } = extractStorageKeyComponents(storageKey);
 
-                // 3. Generate a random startKey, will be overridden if STARTING_KEY_OVERRIDE is set
-                currentStartKey = moduleKey + fnKey + randomHex(paramsKey.length);
-                currentStartKey = STARTING_KEY_OVERRIDE || currentStartKey;
+                let randomEntriesCount = 0;
+                let randomEntries;
+                // Re-try on empty entries cases to avoid false positives
+                while (randomEntriesCount === 0) {
```

Comment:
This may cause an infinite loop, please add something to limit the maximum number of retries.